azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/redux/middleware/handler/CallingMiddlewareActionHandler.kt

```diff
@@ -52,12 +65,31 @@ internal interface CallingMiddlewareActionHandler {
     fun dispose()
     fun admitAll(store: Store<ReduxState>)
     fun admit(userIdentifier: String, store: Store<ReduxState>)
-    fun decline(userIdentifier: String, store: Store<ReduxState>)
+    fun reject(userIdentifier: String, store: Store<ReduxState>)
+    fun removeParticipant(userIdentifier: String, store: Store<ReduxState>)
+    fun setCapabilities(capabilities: Set<ParticipantCapabilityType>, store: Store<ReduxState>)
+    fun onNetworkQualityCallDiagnosticsUpdated(
+        model: CallDiagnosticModel<NetworkCallDiagnostic, CallDiagnosticQuality>,
+        store: Store<ReduxState>
+    )
+    fun onNetworkCallDiagnosticsUpdated(
+        model: CallDiagnosticModel<NetworkCallDiagnostic, Boolean>,
+        store: Store<ReduxState>
+    )
+
+    fun onMediaCallDiagnosticsUpdated(
+        model: CallDiagnosticModel<MediaCallDiagnostic, Boolean>,
+        store: Store<ReduxState>
+    )
+
+    fun dismissNotification(store: Store<ReduxState>)
 }
 
 internal class CallingMiddlewareActionHandlerImpl(
     private val callingService: CallingService,
-    coroutineContextProvider: CoroutineContextProvider
+    coroutineContextProvider: CoroutineContextProvider,
+    private val configuration: CallCompositeConfiguration,
+    private val capabilitiesManager: CapabilitiesManager,
```

Comment:
Can we think about another architecture where we do not inject the manager as a dependency? This would break a lot of unit tests as well.